lua/orgmode/colors/markup_highlighter.lua

```diff
@@ -197,30 +271,77 @@ local function get_matches(bufnr, first_line, last_line)
   local seek_link = {}
   local result = {}
   local link_result = {}
+  local latex_result = {}
 
+  local nested = {}
+  local can_nest = true
   for _, item in ipairs(ranges) do
+    if item.type == '(' then
```

Comment:
Could we have these types in some kind of map? Something like this:
```
  local type_map = {
    ['('] = '\\(',
    [')'] = '\\(',
    ['}'] = '\\{'
  }

  for _, item in ipairs(ranges) do
    item.type = type_map[item.type] or item.type
    -- ...
```

This would remove some of these ifs and elseifs and make code slightly more readable. Of course, for the blocks where we have additional logic we need to have them.

Reply:
Done.